### PR TITLE
Add loading pf filter definitions from file

### PR DESCRIPTION
**Pull request by teknico**
Includes testing and docs. Run the tests with:

```
$ snabb snsh -t apps.lwaftr.conf
```

### CODE REVIEW

File: src/apps/lwaftr/conf_parser.lua [hunk 1]
```diff
@@ -231,6 +231,23 @@ function Parser:parse_string()
...
+   local filter, err = lib.readfile(path, "*a")
+   if filter == nil then
+      self:error('cannot read filter conf file "%s": %s', path, err)
+   end
+   return filter
+end
+
 function Parser:parse_file_name()
    local str = self:parse_string()
    if str == '' then self:error('file name is empty') end
```
Comment (wingo) [inline src/apps/lwaftr/conf_parser.lua]: would you mind refactoring this string-to-file-name resolution stuff into a little helper and call it from both parse_file_name and parse_string_or_file?  thanks :-)

Comment (teknico) [inline src/apps/lwaftr/conf_parser.lua]: Just the one line that does the `lib.dirname(self.name)` concatenation?

Comment (wingo) [inline src/apps/lwaftr/conf_parser.lua]: lines 253 to 258.  thanks

Comment (wingo) [inline src/apps/lwaftr/conf_parser.lua]: which points out a thing!  </foo/bar is totally valid and should be treated as an absolute path.

Comment (teknico) [inline src/apps/lwaftr/conf_parser.lua]: Indeed. That's the case, now that the same logic is used in both places.

**Comment by wingo** [comment]
lgtm if the snabb-bot passes.  thank you @teknico !

**Pull request merged**